### PR TITLE
add docs on profiling including how to make flamegraphs

### PR DESCRIPTION
**Pull request by bjornregnell**
@ekrich Here is an initial proposal for tips and tricks of profiling, based on this discussion: 
https://gitter.im/scala-native/scala-native?at=604f91c9e8267a46f218025e
The proposed doc page "Profiling" is rendered here:
https://github.com/bjornregnell/scala-native/blob/master/docs/user/profiling.rst

### CODE REVIEW

**Comment by teodimoff** [comment]
Cool! This will be valuable for any who wants to debug scala-native.

File: docs/user/profiling.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.. _profiling:
+
+Profiling
+=========
+
+In this section you can find some tips on how to profile your scala-native binary in Linux.
```
Comment (ekrich) [inline docs/user/profiling.rst:6]: I think this line should have _Scala Native_ instead of _scala-native_.

**Review by ekrich — APPROVED** [review]
Other than that small change, LTGM.

File: docs/user/profiling.rst (hunk 1 above)
Comment (bjornregnell) [reply docs/user/profiling.rst]: OK thanks; I have fixed that now and rebased.

**Comment by bjornregnell** [comment]
@ekrich All checks have passed now after rebase+force push upon latest upstream. Ready for merge?

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**